classes/observers/listeners/SendSubmissionToOrcid.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+namespace PKP\observers\listeners;
+
+use APP\facades\Repo;
+use APP\jobs\orcid\DepositOrcidSubmission;
+use APP\orcid\OrcidManager;
+use APP\orcid\OrcidWork;
+use APP\publication\Publication;
+use Carbon\Carbon;
+use Illuminate\Events\Dispatcher;
+use PKP\context\Context;
+use PKP\observers\events\PublicationPublished;
+use PKP\submission\PKPSubmission;
+
+class SendSubmissionToOrcid
+{
+    /**
+     * Maps methods with corresponding events to listen to
+     */
+    public function subscribe(Dispatcher $events): void
+    {
+        $events->listen(
+            PublicationPublished::class,
+            self::class . '@handle'
+        );
+    }
+
+    public function handle(PublicationPublished $publishedEvent): void
+    {
+        $context = $publishedEvent->context;
+        if (!OrcidManager::isEnabled($context)) {
+            return;
+        }
+
+        $publicationStatus = $publishedEvent->publication->getData('status');
+        if ($publicationStatus === PKPSubmission::STATUS_PUBLISHED ||
+                $publicationStatus === PKPSubmission::STATUS_SCHEDULED) {
+            $this->sendSubmissionToOrcid($publishedEvent->publication, $publishedEvent->context);
+        }
+    }
+
+    private function sendSubmissionToOrcid(Publication $publication, Context $context): void
+    {
+        if (!OrcidManager::isMemberApiEnabled($context)) {
+            // Sending to ORCID only works with the member API
+            return;
+        }
+
+        $issueId = $publication->getData('issueId');
+        if (isset($issueId)) {
+            $issue = Repo::issue()->get($issueId);
+        }
+
+        $authors = Repo::author()
+            ->getCollector()
+            ->filterByPublicationIds([$publication->getId()])
+            ->getMany();
+
+        // Collect valid ORCID ids and access tokens from submission contributors
+        $authorsWIthOrcid = [];
+        foreach ($authors as $author) {
+            if ($author->getOrcid() && $author->getData('orcidAccessToken')) {
+                $orcidAccessExpiresOn = Carbon::parse($author->getData('orcidAccessExpiresOn'));
+                if ($orcidAccessExpiresOn->isFuture()) {
+                    # Extract only the ORCID from the stored ORCID uri
+                    $orcid = basename(parse_url($author->getOrcid(), PHP_URL_PATH));
+                    $authorsWithOrcid[$orcid] = $author;
+                } else {
+                    // TODO: In sandbox mode, the `DepositOrcidSubmission` job is dispatched but is not actually run.
+                    //      Is it okay to still remove the expired ORCID access tokens even if the job isn't being run?
+                    //      This shouldn't be a problem as they're already expired, but even so...
```

Comment:
Having looked at it in more detail, I don't think this will be a problem. With the OJS sandbox mode, the job will "run" but it will exit early. We could fail the job and let it retry automatically. At the very least there would be a record of the job failing if they needed to be rerun.

As for the journal manager switching to or from the type of ORCID API, the permissions are associated with the individual access token, so if something has changed (e.g. a journal no longer being an ORCID member and not having access to the ORCID member API, the request would fail as if it were unauthenticated).

A long-winded way to say I think it should be okay as-is, but a good thing to keep in mind for in testing pre-release.